app/experimenter/nimbus-ui/src/lib/dateUtils.ts

```diff
@@ -48,6 +48,26 @@ export function stringDateSubtract(dateA: string, dateB: string): string {
   return pluralize(Math.round(dateDifference / DAY), "day");
 }
 
+/**
+ *  Results are available if enrollment is complete and
+ *  more than a week has passed after that.
+ */
+export function resultsReady(
+  experiment: getExperiment_experimentBySlug | getAllExperiments_experiments,
+): boolean {
+  const DAYS_UNTIL_ANALYSIS = 8;
+  const today = new Date();
+  const { startDate, proposedEnrollment } = experiment;
+  const experimentStartDate = new Date(String(startDate));
+
+  const resultsReadyDate = new Date();
+  resultsReadyDate.setDate(
+    experimentStartDate.getDate() + proposedEnrollment + DAYS_UNTIL_ANALYSIS,
+  );
+
+  return today.getDate() >= resultsReadyDate.getDate();
+}
+
```

Comment:
You could do this on the backend, particularly if there's a constant involved.  All the constants right now live in the NimbusConstants file in Python, so it might get confusing later if we have a constant hidden off in the frontend?

Reply:
Sure we can move this to the backend

Reply:
Actually, I should note that the reason I didn't initially do it on the backend in the visualization API is because it takes some time to load the visualization data and I don't want to do that in the home page just to check if results are available.

So this logic would have to attach a field to the `experiment` object

Reply:
Yeah exactly, make it a method on the NimbusExperiment class to look up the dates and compare to the constant since there's already lots of methods in there that do that same kind of thing and then attach it as a query property in the NimbusExperiment query type.